app/renderer/components/Session/FormattedCaps.js

```diff
@@ -17,9 +17,36 @@ export default class FormattedCaps extends Component {
   }
 
   render () {
-    const {caps, title, isEditingDesiredCaps, startDesiredCapsEditor, abortDesiredCapsEditor, saveRawDesiredCaps, setRawDesiredCaps, rawDesiredCaps,
+    const {caps, title, isEditingDesiredCapsName, startDesiredCapsNameEditor, abortDesiredCapsNameEditor, saveRawDesiredCapsName, setRawDesiredCapsName, rawDesiredCapsName,
+           isEditingDesiredCaps, startDesiredCapsEditor, abortDesiredCapsEditor, saveRawDesiredCaps, setRawDesiredCaps, rawDesiredCaps,
            isValidCapsJson, invalidCapsJsonReason, t} = this.props;
-    return caps && <Card title={title || 'JSON Representation'} className={SessionStyles.formattedCaps}>
+    return caps && <Card className={SessionStyles.formattedCaps}
+      title={!title ? 'JSON Representation' : (!isEditingDesiredCapsName ? title :
```

Comment:
for readability, rather than using a nested ternary i would break all this out into a set of if/then statements above culminating in one variable that can be assigned to `title` here. the same strategy might also improve the readability of the sections below.